src/patchkernel/patch_kernel_parallel.cpp

```diff
@@ -208,12 +208,13 @@ bool PatchKernel::isDistributed(bool allowDirty) const
 */
 int PatchKernel::getOwner(bool allowDirty) const
 {
-	assert(allowDirty || arePartitioningInfoDirty(false));
-	if (!allowDirty || !arePartitioningInfoDirty(true)) {
-		return m_owner;
-	} else {
+	if (allowDirty) {
 		return evalOwner();
 	}
+
+	assert(!arePartitioningInfoDirty(false));
```

Comment:
Another booby trap. If you're working  MPI with 1 processor, isPartitioned is always true (that means communicator is always set by class constructor), arePartitioningInfoDirty return always true. See below